DOCUMENTATION.md

```diff
@@ -0,0 +1,7 @@
+# Documentation of the changes made
+
+In the refactoring of this code, the first thing I noticed was the fact that there was no protected module. The protected module keeps methods from being mistakenly called. So, the first thing I did was to add the protected module. I also noticed that the articleParams method was improperly named so I renamed it article_params everywhere it was called.
+
+Next, I noticed that the index method was too long. I resolved to take out articles_type and articles_boundary. The articles_boundary was also used in the feed method so I took out the similar lines contained therein. I also removed the arrow from the code in the create method.
```

Comment:
Yes, a controller action should be solely responsible for receiving requests , process the request and then produce a response.  So this is totally on point.

However, the controller action should not care about the details of how the request is processed. It should only care about receiving the request and then the giving out a response.

So this methods you created here should be moved into an abstraction. By abstraction here, I mean either the `Article` class or a new abstraction

Reply:
Right. I should have moved the methods I brought out to the `protected` part. I did not move it out of the file because we have only one file we are dealing with.